bot/common.py

```diff
@@ -313,13 +315,13 @@ def entity_type_loop(bot, entitytype, limit):
     linkids = const.LINK_IDS[entitytype]
 
     wiki_entity_query = create_url_mbid_query(entitytype, linkids)
-    all_results = do_readonly_query(wiki_entity_query, limit)
     already_processed_query = create_already_processed_query(entitytype)
-    already_processed_results = frozenset(
-           do_readwrite_query(already_processed_query))
 
-    results_to_process = [r for r in all_results if r[0] not in
-                          already_processed_results]
+    with do_readonly_query(wiki_entity_query, limit) as all_results, do_readwrite_query(already_processed_query) as already_processed:
+        already_processed_results = frozenset(already_processed)
+
+        results_to_process = [r for r in all_results if r[0] not in
+                              already_processed_results]
 
     if len(results_to_process) == 0:
```

Comment:
`results_to_process` may not be defined here, and list (non-)emptyness can be tested just with `if results_to_process:`
You may want to move this code inside the `with` statement:

```python
with do_readonly_query(wiki_entity_query, limit) as all_results, do_readwrite_query(already_processed_query) as already_processed:
    results_to_process = [r for r in all_results if r[0] not in frozenset(already_processed)]
    if results_to_process:
        map(bot.process_result, results_to_process)
    else:
        wp.output("No more unprocessed entries in MB")
```

Also the `map()` may be just replaced by a `for` loop + list comprehensions.

Reply:
`results_to_process` will always be defined at that point.

Moving the `bot.process_result` calls into the `with` statements block defeats the whole purpose of this change - reducing the time the cursors from `do_*_query` are open.